Assets/MixedRealityToolkit.Tests/PlayModeTests/InteractableTests.cs

```diff
@@ -762,6 +763,100 @@ public IEnumerator TestInteractableToggleCollectionIndexUpdate()
+            Assert.False(interactable.HasPress);
+            Assert.False(interactable.HasGesture);
+            Assert.True(interactable.StateManager.CurrentState().Name == "Focus", "Interactable State is not Focus");
+
+
+            // Pinch Gesture and keeping focus
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+
+            Assert.True(interactable.HasFocus);
+            Assert.True(interactable.HasPress);
+            Assert.False(interactable.HasGesture);
+            Assert.True(interactable.StateManager.CurrentState().Name == "Pressed", "Interactable State is not Pressed");
+
+
+            // Move Hand to make button out of focus. Button State should go to Default State
+
+            yield return rightHand.Move(releaseDelta);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+
+            Assert.False(interactable.HasFocus, "focus");
+            Assert.False(interactable.HasPress, "press");
+            Assert.False(interactable.HasGesture, "gesture");
+            Assert.True(interactable.StateManager.CurrentState().Name == "Default", "Interactable State is not Default");
+
+
+            // Open hand. Button should stay on Default State
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+
+            Assert.False(interactable.HasFocus, "focus");
+            Assert.False(interactable.HasPress, "press");
+            Assert.False(interactable.HasGesture, "gesture");
+            Assert.True(interactable.StateManager.CurrentState().Name == "Default", "Interactable State is not Default");
+
+
+            // Move Hand back to Initial position and Pinch. Button should go to Pressed State
+
+            yield return rightHand.Move(-releaseDelta);
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Pinch);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+
+            Assert.True(interactable.HasFocus, "focus");
+            Assert.True(interactable.HasPress, "press");
+            Assert.False(interactable.HasGesture, "gesture");
+            Assert.True(interactable.StateManager.CurrentState().Name == "Pressed", "Interactable State is not Pressed");
+
+
+            // Open Hand. Button should go to Focus State
+
+            yield return rightHand.SetGesture(ArticulatedHandPose.GestureId.Open);
+            yield return PlayModeTestUtilities.WaitForInputSystemUpdate();
+            Assert.True(interactable.HasFocus, "focus");
+            Assert.False(interactable.HasPress, "press");
+            Assert.False(interactable.HasGesture, "gesture");
+            Assert.True(interactable.StateManager.CurrentState().Name == "Focus", "Interactable State is not Focus");
+
+
+            //Cleanup
```

Comment:
```suggestion
            // Cleanup
```

Reply:
A meta comment on comments in general, because I see some really awesome comments in here (and want to encourage those) and some comments that have a little less value (i.e. // Cleanup). The goal of me leaving this here isn't to discourage comments, but hopefully encourage more instances of helpful comments over other types.

https://testing.googleblog.com/2017/07/code-health-to-comment-or-not-to-comment.html

There are some times where code is somewhat self documenting (i.e. this case where Destroy is obviously cleaning things up), and in those cases having a simple comment doesn't necessarily add that much. The linked blog entry has a lot of good cases where comments are extremely valuable (i.e. intent - and history and why). Of course even those have some *challenges* in that intent in comments can obviously get out of date, but overall I've found it's added more value over time than caused issues.

If this comment were instead like "Cleanup the test's Interactable to ensure that other tests are not affected by this tests's state changes" then it might also cover a little more intent behind the thingy.

Reply:
That's my bad. I had clean up added I believe in other parts of the tests. It was likely copied/pasted

It should be removed

Reply:
Makes sense.